subsys/mgmt/mcumgr/grp/settings_mgmt/src/settings_mgmt.c

```diff
@@ -460,9 +526,35 @@ static int settings_mgmt_save(struct smp_streamer *ctxt)
 		}
 	}
 
-	settings_save();
+	if (save_one) {
+		rc = settings_save_one(key_name, data.value, data.len);
+	} else if (save_subtree) {
+		rc = settings_save_subtree(key_name);
+	} else {
+		rc = settings_save();
+	}
+
+	switch (rc) {
+	case 0:
+		break;
+	case -ENOENT:
+	case -ENOTSUP:
+		rc = SETTINGS_MGMT_ERR_SAVE_NOT_SUPPORTED;
+		break;
+	default:
+		rc = SETTINGS_MGMT_ERR_UNKNOWN;
+		break;
+	}
+
+	if (rc != 0) {
+		ok = smp_add_cmd_err(zse, MGMT_GROUP_ID_SETTINGS, (uint16_t)rc);
+	}
```

Comment:
```suggestion
	if (rc != 0) {
		switch (rc) {
		case -ENOENT:
		case -ENOTSUP:
			rc = SETTINGS_MGMT_ERR_SAVE_NOT_SUPPORTED;
			break;
		default:
			rc = SETTINGS_MGMT_ERR_UNKNOWN;
			break;
		}
		
		ok = smp_add_cmd_err(zse, MGMT_GROUP_ID_SETTINGS, (uint16_t)rc);
	}
```

to keep style as other functions